### PR TITLE
point non-dashboard ICDS ucrs to new DB

### PR DESCRIPTION
**Pull request by sravfeyn**
@snopoke @calellowitz @emord

### CODE REVIEW

**Comment by sravfeyn** [comment]
To go with https://github.com/dimagi/commcare-cloud/pull/2623/files

**Comment by sravfeyn** [comment]
There are [various other places](https://github.com/dimagi/commcare-hq/search?q=get_icds_ucr_db_alias&unscoped_q=get_icds_ucr_db_alias) `get_icds_ucr_db_alias` is used that I assume we fully don't need to worry about. I have only modified the db migrations to run on new db.

**Comment by sravfeyn** [comment]
I have tested this locally by creating a new db locally for the non-dashboard UCRs with above setup and it works correctly.

**Review by calellowitz — APPROVED** [review]

File: corehq/sql_db/routers.py [hunk 1]
```diff
@@ -55,7 +56,8 @@ def allow_migrate(db, app_label):
     """
     if app_label == ICDS_REPORTS_APP:
         db_alias = get_icds_ucr_db_alias()
-        return bool(db_alias and db_alias == db)
+        non_dashboard_alias = get_icds_ucr_non_dashboard_db_alias()
+        return bool(db_alias and db_alias == db) or bool(non_dashboard_alias and non_dashboard_alias == db)
```
Comment (emord) [inline corehq/sql_db/routers.py]: Is this necessary? I think that all the UCR stuff will be handled by SQLAlchemy and this is just for django routing right?

Comment (sravfeyn) [reply corehq/sql_db/routers.py]: This is to determine if any migrations defined in ICDS_REPORTS_APP should be run on this new DB too or not. I assumed there would be some django migrations that might need to be run, but if that's not true, I can remove it.

Comment (sravfeyn) [reply corehq/sql_db/routers.py]: Removed this as well, per the [comment](https://github.com/dimagi/commcare-cloud/pull/2623#discussion_r255578821)

**Comment by sravfeyn** [comment]
@emord is this g2g?

File: custom/icds_reports/ucr/data_sources/ls_home_visit_forms_filled.json [hunk 2]
```diff
@@ -173,7 +173,7 @@
       }
     },
     "named_filters": {},
-    "engine_id": "icds-ucr",
+    "engine_id": "icds-ucr-non-dashboard",
```
Comment (snopoke) [inline custom/icds_reports/ucr/data_sources/ls_home_visit_forms_filled.json]: I think this one can't be moved (it's not in the list on the trello card)

**Review by snopoke — APPROVED** [review]

**Review by emord — APPROVED** [review]

**Pull request merged**